Python/ceval.c

```diff
@@ -960,6 +960,13 @@ _PyEval_EvalFrameDefault(PyThreadState *tstate, _PyInterpreterFrame *frame, int
     if ((COND)) {                \
         goto deoptimize;\
     }
+#undef TIER2_DEOPT_IF
+
+#define TIER2_DEOPT_IF(COND, INSTNAME)    \
+    if ((COND)) {                         \
+        next_uop = current_executor->trace + (next_uop-1)->target;            \
```

Comment:
I believe @brandtbucher objected to using `next_uop[-1].operand` since it complicates the copy-and-patch JIT code generation; presumably this also isn't going to make him happy.

(But if this code survives, please fix the whitespace.)

Reply:
Yeah, we currently do `next_uop[-1].operand` in the `executor_cases.c.h`

Reply:
Po-tay-to, po-tah-to, the point is that the Justin branch doesn’t like to reference ‘next_uop’.

Reply:
`operand` is a problem (since we want to burn it into the generated code), but this is actually fine. We don’t care where or how the code is jumping in the uop interpreter, since we have a completely different mechanism for jumping in the jitted code. The JIT template can just define this macro differently.

Reply:
> `operand` is a problem (since we want to burn it into the generated code), but this is actually fine. We don’t care where or how the code is jumping in the uop interpreter, since we have a completely different mechanism for jumping in the jitted code. The JIT template can just define this macro differently (likely to jump to a deopt label, as it does now).

Hm, but if the target field is used to jump to a different uop in the current trace, the jitter would have to change to make each deopt jump to the right target, wouldn't it? IIUC the jitter currently translates all DEOPT_IF() calls to `goto deoptimize` where that is a fixed label that makes a tail call to `_JIT_DEOPTIMIZE`. But with `target`, it would instead have to tail call the target uop.

Reply:
Yep! But that's fine. We can leave `DEOPT_IF` to jump to the label. `_JIT_DEOPTIMIZE` is patched in by the JIT compiler, so we can just as easily patch it to be wherever `target` happens to be for that instruction. Easy-peasy.